src/services/logger.service.ts

```diff
@@ -0,0 +1,32 @@
+import Winston from 'winston';
+
+export class LoggerService {
+  private _loggerInstance: Winston.Logger;
+
+  public constructor() {
+    this._loggerInstance = Winston.createLogger({});
+
+    this._loggerInstance = Winston.createLogger({
+      level: 'info',
+      format: Winston.format.json(),
+      defaultMeta: { service: 'lion' },
+      transports: [
+        new Winston.transports.File({ filename: 'error.log', level: 'error' }),
+        new Winston.transports.File({ filename: 'combined.log' }),
+      ],
+      exceptionHandlers: [new Winston.transports.File({ filename: 'exceptions.log' })],
+    });
+
+    if (process.env.NODE_ENV !== 'production') {
+      this._loggerInstance.add(
+        new Winston.transports.Console({
+          format: Winston.format.simple(),
+        })
+      );
+    }
+  }
+
+  public get() {
```

Comment:
instead of exposing the logger instance to the client, create methods such as `.info()`, `.warn()`, etc that wrap around the logger instance. it isn't good to expose the instance to the consumer incase we decide to move away from winston for some reason

Reply:
also, eventually i will want to push these logs to some 3rd party place so we can view production logs in realtime. we wouldn't be able to implement that with this current implementation.

Reply:
there are a ton of things like `info`/`warn` :(

Reply:
re: realtime logs, they get published to files; but we can add things to the winston logger so that they also publish elsewhere.